### ISSUE TITLE
[Feature Request] Windows Explorer integration / shell integration

### ISSUE THREAD

**Issue by lukeawyatt** [ForkIssues/TrackerWin#89]
Good work on this!  I'm currently using version 1.25.1.0 on Windows and have freshly found this project.  I think some shell integration could definitely help user workflow.  The expected behavior would be:

1. User right-clicks on a folder or in a folder using Explorer
2. "Open with Fork" appears in the context menu
3. User clicks the context menu item
4. Fork opens, loads an existing repo if it exists, or prompts to initialize a new repo if not

Let me know your thoughts
Thanks, Luke

**Comment by Otiel** [comment]
If this is implemented, please add an option to disable it. I'd prefer to keep my context menu clean.

**Comment by glitch-bane** [comment]
I'd imagine it'd go in as an option during install if so.  Similar to how Git-SCM adds Git Bash to the context menu.

**Comment by ravindUwU** [comment]
IMO, this should be implemented the way 7-zip does it, where settings page of the app provides options to toggle/customize the context menu,

![7-zip Context Menu Customization](https://user-images.githubusercontent.com/7352580/50738388-4e517180-11f9-11e9-83a2-7452ff1c71aa.PNG)

**Comment by DavorLovric** [comment]
I found this while looking for `Open with Fork` context menu action.
Being impatient, I looked for the solution, at least for Win10 (I haven't tested on other versions). It turned out that the context menu is defined in the registry and it can be modified.

1. Create `AddOpenWithFork.reg` text file and put the following text in it:
```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\Directory\Background\shell\Fork]
@="Open with Fork"
"Icon"="C:\\Users\\<username>\\AppData\\Local\\Fork\\Fork.exe"

[HKEY_CLASSES_ROOT\Directory\Background\shell\Fork\command]
@="\"C:\\Users\\<username>\\AppData\\Local\\Fork\\Fork.exe\" \"%V\""
```
2. Replace `<username>` with your username.
3. Save and close the file.
4. Right click on the file and select Merge.

Your context menu now has "Open with Fork",

**Comment by quotidian-ennui** [comment]
You *could* just abuse the SendTo menu (rather than messing with the registry); Find the fork start-menu item; copy it and, paste it into `C:\Users\<username>\AppData\Roaming\Microsoft\Windows\SendTo`

Then your send-to menu has fork so you can right-click on a directory -> sendTo -> fork.  I think that fork definitely needs a single instance option as per #266

**Comment by Lonli-Lokli** [comment]
Yep, I definitely missing feature like 'View file history' in context menu of Explorer

**Comment by Grahamvs** [comment]
> 
> 
> I found this while looking for `Open with Fork` context menu action.
> Being impatient, I looked for the solution, at least for Win10 (I haven't tested on other versions). It turned out that the context menu is defined in the registry and it can be modified.
> 
>     1. Create `AddOpenWithFork.reg` text file and put the following text in it:
> 
> 
> ```
> Windows Registry Editor Version 5.00
> 
> [HKEY_CLASSES_ROOT\Directory\Background\shell\Fork]
> @="Open with Fork"
> "Icon"="C:\\Users\\<username>\\AppData\\Local\\Fork\\Fork.exe"
> 
> [HKEY_CLASSES_ROOT\Directory\Background\shell\Fork\command]
> @="\"C:\\Users\\<username>\\AppData\\Local\\Fork\\Fork.exe\" \"%V\""
> ```
> 
>     1. Replace `<username>` with your username.
> 
>     2. Save and close the file.
> 
>     3. Right click on the file and select Merge.
> 
> 
> Your context menu now has "Open with Fork",

Just to add to this, it is do-able via command using the following:

```
REG ADD HKCR\Directory\Background\shell\Fork /d "Open with Fork" /t REG_SZ /f
REG ADD HKCR\Directory\Background\shell\Fork /v Icon /d "C:\\Users\\%username%\\AppData\\Local\\Fork\\Fork.exe" /t REG_SZ /f
REG ADD HKCR\Directory\Background\shell\Fork\command /d "\"C:\\Users\\%username%\\AppData\\Local\\Fork\\Fork.exe\" \"\%V\"" /t REG_SZ /f
```

This approach should auto-resolve the username.

**Comment by trueaniki** [comment]
> I found this while looking for `Open with Fork` context menu action.
> Being impatient, I looked for the solution, at least for Win10 (I haven't tested on other versions). It turned out that the context menu is defined in the registry and it can be modified.
> 
> 1. Create `AddOpenWithFork.reg` text file and put the following text in it:
> 
> ```
> Windows Registry Editor Version 5.00
> 
> [HKEY_CLASSES_ROOT\Directory\Background\shell\Fork]
> @="Open with Fork"
> "Icon"="C:\\Users\\<username>\\AppData\\Local\\Fork\\Fork.exe"
> 
> [HKEY_CLASSES_ROOT\Directory\Background\shell\Fork\command]
> @="\"C:\\Users\\<username>\\AppData\\Local\\Fork\\Fork.exe\" \"%V\""
> ```
> 
> 1. Replace `<username>` with your username.
> 2. Save and close the file.
> 3. Right click on the file and select Merge.
> 
> Your context menu now has "Open with Fork",

I want to add, that it is possible to add Fork to directory context menu not in background, just add simular keys to HKEY_CLASSES_ROOT\Directory\shell\Fork, but replace "%V" in @="\"C:\\Users\\<username>\\AppData\\Local\\Fork\\Fork.exe\" \"%V\"" with "%1"

**Comment by belaw** [comment]
Generic .reg file that uses the %LOCALAPPDATA% environment variable. No manual edits required.
https://gist.github.com/belaw/3bf516e14f04b9263cd846e6b85dfec6
